pkg/network/setup/netstat.go

```diff
@@ -67,106 +66,78 @@ func (c *NetStat) CachePodInterfaceVolatileData(vmi *v1.VirtualMachineInstance,
+	if len(domain.Status.Interfaces) > 0 {
+		guestAgentInterfacesByMac := map[string]api.InterfaceStatus{}
 		for _, domainInterfaceStatus := range domain.Status.Interfaces {
-			domainInterfaceStatusByMac[domainInterfaceStatus.Mac] = domainInterfaceStatus
+			guestAgentInterfacesByMac[domainInterfaceStatus.Mac] = domainInterfaceStatus
 		}
 
-		existingInterfacesSpecByName := map[string]v1.Interface{}
+		vmiSpecInterfacesByName := map[string]v1.Interface{}
 		for _, existingInterfaceSpec := range vmi.Spec.Domain.Devices.Interfaces {
-			existingInterfacesSpecByName[existingInterfaceSpec.Name] = existingInterfaceSpec
+			vmiSpecInterfacesByName[existingInterfaceSpec.Name] = existingInterfaceSpec
 		}
-		existingNetworksByName := map[string]v1.Network{}
-		for _, existingNetwork := range vmi.Spec.Networks {
-			existingNetworksByName[existingNetwork.Name] = existingNetwork
-		}
-
-		// Iterate through all domain.Spec interfaces
-		for _, domainInterface := range domain.Spec.Devices.Interfaces {
-			interfaceMAC := domainInterface.MAC.MAC
-			var newInterface v1.VirtualMachineInstanceNetworkInterface
-			var isForwardingBindingInterface = false
-
-			if existingInterfacesSpecByName[domainInterface.Alias.GetName()].Masquerade != nil || existingInterfacesSpecByName[domainInterface.Alias.GetName()].Slirp != nil {
-				isForwardingBindingInterface = true
-			}
 
-			if existingInterface, exists := existingInterfaceStatusByName[domainInterface.Alias.GetName()]; exists {
-				// Reuse previously calculated interface from vmi.Status.Interfaces, updating the MAC from domain.Spec
-				// Only interfaces defined in domain.Spec are handled here
-				newInterface = existingInterface
-				newInterface.MAC = interfaceMAC
-
-				// If it is a Combination of Masquerade+Pod network, check IP from file cache
-				if existingInterfacesSpecByName[domainInterface.Alias.GetName()].Masquerade != nil && existingNetworksByName[domainInterface.Alias.GetName()].NetworkSource.Pod != nil {
-					iface, err := c.getPodInterfacefromFileCache(vmi, domainInterface.Alias.GetName())
-					if err != nil {
-						return err
-					}
-
-					if !reflect.DeepEqual(iface.PodIPs, existingInterfaceStatusByName[domainInterface.Alias.GetName()].IPs) {
-						newInterface.Name = domainInterface.Alias.GetName()
-						newInterface.IP = iface.PodIP
-						newInterface.IPs = iface.PodIPs
-					}
-				}
-			} else {
-				// If not present in vmi.Status.Interfaces, create a new one based on domain.Spec
-				newInterface = v1.VirtualMachineInstanceNetworkInterface{
-					MAC:  interfaceMAC,
-					Name: domainInterface.Alias.GetName(),
+		// Override status data with information from the guest-agent
+		for index := range vmi.Status.Interfaces {
+			vmiStatusInterface := &vmi.Status.Interfaces[index]
+			if guestAgentInterface, exists := guestAgentInterfacesByMac[vmiStatusInterface.MAC]; exists {
+				delete(guestAgentInterfacesByMac, vmiStatusInterface.MAC)
```

Comment:
Why is it necessary to delete the item from `guestAgentInterfacesByMac` map?

Reply:
On line 140, the interfaces that have no been processed here have special treatment.
I left this part as it original was because I did not have something better (or more accurately, it was not the problematic part that I tried to solve).

I guess I can invest some time and see how it can be eliminated.